docs/content/ecs/component-generation.md

```diff
@@ -13,27 +13,49 @@ The [code generator]({{urlRoot}}/content/code-generator) uses `.schema` files to
 
 ## Overview
 
-A `struct`, which implements `Unity.Entities.IComponentData` and `Improbable.Gdk.Core.ISpatialComponentData`, is generated for each SpatialOS component. It has the same name as the corresponding schema component.
-The generation process creates a namespace for each struct according to the relevant schemalang component name: `{name of schema component}.Component`
+The code generator generates two structs for each SpatialOS component defined in schema. The generation process places these structs in a namespace according to the SpatialOS component name: `{schema package}.{name of schema component}`. The two structs it generates are:
+
+  * A `struct` which implements `Improbable.Gdk.Core.ISpatialComponentSnapshot`. Its fully qualified name is: `{schema package}.{name of schema component}.Snapshot`.
+  * A `struct` which implements `Unity.Entities.IComponentData`, `Improbable.Gdk.Core.ISpatialComponentData`, and `Improbable.Gdk.Core.ISnapshottable<{schema package}.{name of schema component}.Snapshot>`. Its fully qualified name is: `{schema package}.{name of schema component}.Component`.
+
 These structs only contain the defined schema data fields. They do *not* contain any fields or methods relating to [commands]({{urlRoot}}/content/ecs/sending-receiving-component-commands) or [events (SpatialOS documentation)](https://docs.improbable.io/reference/latest/shared/glossary#event) defined on that component.
 
 ## Generation details
-Each struct contains the following fields:
+
+### Snapshot
+
+This struct contains the following fields:
+
+  * the public property `uint ComponentId` to read the component ID of this component as defined in schemalang
```

Comment:
```suggestion
  * the public property `uint ComponentId` to read the component ID of this component as defined in [schemalang (SpatialOS documentation)](https://docs.improbable.io/reference/latest/shared/glossary#schemalang)
```

Reply:
Good idea to link to what this means - we don't have it in the local glossary but I think it should be in there, really. Will add to my non-urgent TODOs. (the list is growing.)

Reply:
Link to what? The component ID ?